packages/@aws-cdk/aws-codebuild/lib/source.ts

```diff
@@ -106,6 +105,67 @@ export abstract class GitBuildSource extends BuildSource {
   }
 }
 
+/**
+ * The construction properties common to all external build sources that are backed by Git.
+ */
+export interface ExternalGitBuildSourceProps extends GitBuildSourceProps {
+  /**
+   * Whether to create a webhook that will trigger a build every time a commit is pushed to the repository.
+   *
+   * @default false
+   */
+  readonly webhook?: boolean;
+
+  /**
+   * Whether to send notifications on your build's start and end.
+   *
+   * @default true
+   */
+  readonly reportBuildStatus?: boolean;
+
+  /**
+   * A list of lists of WebhookFilter objects used to determine which webhook events are triggered.
+   */
+  readonly filterGroups?: WebhookFilter[][];
```

Comment:
Why a list of lists? What are the semantics here?

Reply:
It's an OR operation between different lists and an AND operation inside the list.
For example, if I have filter groups `[[A,B],[C,D]]`, build will only be triggered if `(A&B)|(C&D)`